hyperspy/_signals/sed.py

```diff
@@ -0,0 +1,465 @@
+            sed_par.scan_rotation,
+            'Acquisition_instrument.TEM.convergence_angle':
+            sed_par.convergence_angle,
+            'Acquisition_instrument.TEM.precession_angle':
+            sed_par.precession_angle,
+            'Acquisition_instrument.TEM.precession_frequency':
+            sed_par.precession_frequency,
+            'Acquisition_instrument.TEM.Detector.SED.exposure_time':
+            sed_par.exposure_time, }
+
+        for key, value in mapping.iteritems():
+            if value != t.Undefined:
+                self.metadata.set_item(key, value)
+        self._are_microscope_parameters_missing()
+
+    def _are_microscope_parameters_missing(self):
+        """Check that the SED parameters necessary for pattern calibration are
+        defined in metadata and raise a UI if not to add them."""
+        must_exist = ('Acquisition_instrument.TEM.beam_energy',
+                      'Acquisition_instrument.TEM.camera_length',
+                      'Acquisition_instrument.TEM.scan_rotation')
+
+        missing_parameters = []
+        for item in must_exist:
+            exists = self.metadata.has_item(item)
+            if exists is False:
+                missing_parameters.append(item)
+        if missing_parameters:
+            if preferences.General.interactive is True:
+                par_str = "The following parameters are missing:\n"
+                for par in missing_parameters:
+                    par_str += '%s\n' % par
+                par_str += 'Please set them in the following wizard'
+                is_ok = messagesui.information(par_str)
+                if is_ok:
+                    self._set_microscope_parameters()
+                else:
+                    return True
+            else:
+                return True
+        else:
+            return False
+
+    def get_direct_beam_mask(self, radius=None, center=None):
+        """Generate a signal mask for the direct beam.
+
+        Parameters
+        ----------
+        radius : int
+            User specified radius for the circular mask.
+        center : tuple, None
+            User specified (x, y) position of the diffraction pattern center.
+            i.e. the direct beam position. If None it is assumed that the direct
+            beam is at the center of the diffraction pattern.
+
+        Return
+        ------
+        mask : array
+            The mask of the direct beam
+        """
+        r = radius
+        ny = self.axes_manager.signal_shape[1]
+        nx = self.axes_manager.signal_shape[0]
+
+        if center == None:
+            a = (self.axes_manager.signal_shape[1] - 1) / 2
+            b = (self.axes_manager.signal_shape[0] - 1) / 2
+        else:
+            a, b = center
+
+        y, x = np.ogrid[-b:ny-b, -a:nx-a]
+        mask = x*x + y*y <= r*r
+        return mask
+
+    def get_direct_beam_position(self, radius=None):
+        """Estimate the position of the direct beam in each SED pattern.
+
```

Comment:
If I understood well the algorithm it:

1. Estimates an average direct beam position as the average position of the saturated area.
2. Scans through all images and only if in the image the position of the average direct beam position is not saturated recomputes the direct beam position just for this image using the same method as in 1.

If this is correct I have the following comments:

* Why 2. has to go in a while loop: it'll never perform more than 1 iteration.
* Why not calculating the position for each image independently? This should be more accurate as recomputing only when the estimation is so bad that it is out of the saturated area seems like not aiming at the best possible accuracy.
*  For big data, it would probably be better to skip the average position estimate as doing this in a huge dataset is expensive. Instead, it could:
    1. calculate it for the first image as a whole
    2. Calculate it in a ROI around the position of the first image for the following DPs. If the saturated value is not included in the ROI, take the whole image again.
* Why not estimating the radius if not given?